odoo/tests/common.py

```diff
@@ -718,6 +722,21 @@ def setUpClass(cls):
 
         cls.cr = cls.registry.cursor()
         cls.addClassCleanup(cls.cr.close)
+        if cls.rollback_sequences:
+            cls.cr.execute("SELECT sequencename, last_value FROM pg_sequences")
+            cls._starting_sequences = {
+                sequence: last_value
+                for sequence, last_value in cls.cr.fetchall()
+                if sequence not in NON_ROLLBACKED_SEQUENCES
+            }
+
+            @cls.addClassCleanup
+            def _reset_sequences():
+                cls.cr.execute("SELECT sequencename, last_value FROM pg_sequences")
+                for sequence, last_value in cls.cr.fetchall():
+                    start_value = cls._starting_sequences.get(sequence, last_value)
+                    if start_value != last_value:
+                        cls.cr.execute('SELECT setval(%s, %s)', (sequence, start_value))
```

Comment:
That is a very good recipe to break concurrent transactions with race conditions, since sequences are cross-transactional. What guarantees do we have there will be none? 🤔

Reply:
None, this is why it is only opt-in for local testing.